src/main/kotlin/no/nav/familie/ef/søknad/api/dto/pdl/Barn.kt

```diff
@@ -0,0 +1,60 @@
+package no.nav.familie.ef.søknad.api.dto.pdl
+
+import java.time.LocalDate
+
+class Barn(val fnr: String,
+           _navn: String,
+           val alder: Int,
+           val fødselsdato: LocalDate,
+           val harSammeAdresse: Boolean,
+           val medForelder: MedForelder?,
+           val harAdressesperre: Boolean
+) {
+
+    val navn: String
+
+    init {
+        this.navn = when (harAdressesperre) {
+            true -> ""
+            false -> _navn
+        }
+
+    }
+
+
+    override fun equals(other: Any?): Boolean {
+        if (this === other) return true
+        if (javaClass != other?.javaClass) return false
+
+        other as Barn
+
+        if (fnr != other.fnr) return false
+        if (alder != other.alder) return false
+        if (fødselsdato != other.fødselsdato) return false
+        if (harSammeAdresse != other.harSammeAdresse) return false
+        if (medForelder != other.medForelder) return false
+        if (harAdressesperre != other.harAdressesperre) return false
+        if (navn != other.navn) return false
+
+        return true
+    }
+
+    override fun hashCode(): Int {
+        var result = fnr.hashCode()
+        result = 31 * result + alder
+        result = 31 * result + fødselsdato.hashCode()
```

Comment:
Oi, long-time-no-see på equals/hashcode-override. Hva er årsaken? Pga overriding av navn?

Reply:
Ja, og kontroll på toString. Tar gjerne en prat om dette.

Reply:
Du trenger vel ikke ovveride equals, og hash for å override toString?